cmd/osquery-perf/agent.go

```diff
@@ -1292,26 +1383,20 @@ func (a *agent) setMDMEnrolled() {
 }
 
 func (a *agent) mdmWindows() []map[string]string {
-	autopilot := rand.Intn(2) == 1
-	ix := rand.Intn(len(possibleMDMServerURLs))
-	serverURL := possibleMDMServerURLs[ix]
-	providerID := fleet.MDMNameFromServerURL(serverURL)
-	installType := "Microsoft Workstation"
-	if rand.Intn(4) == 1 {
-		installType = "Microsoft Server"
-	}
-
-	rows := []map[string]string{
-		{"key": "discovery_service_url", "value": serverURL},
-		{"key": "installation_type", "value": installType},
-	}
-	if providerID != "" {
-		rows = append(rows, map[string]string{"key": "provider_id", "value": providerID})
+	if !a.mdmEnrolled() {
+		return []map[string]string{
+			// empty service url means not enrolled
+			{"is_federated": "0", "discovery_service_url": "", "provider_id": "", "installation_type": "Client"},
+		}
 	}
-	if autopilot {
-		rows = append(rows, map[string]string{"key": "autopilot", "value": "true"})
+	return []map[string]string{
+		{
+			"is_federated":          "0",
+			"discovery_service_url": a.serverAddress,
+			"provider_id":           fleet.WellKnownMDMFleet,
```

Comment:
Same here, I think that "provider_id" ensures this is recognized as Fleet MDM.

Reply:
That seems correct so long as the `discovery_server_url` is not an [empty string](https://github.com/fleetdm/fleet/blob/fdcf255d6caf59cc8a03fc2023a023ac73d565bf/server/service/osquery_utils/queries.go#L1555-L1556) (which should not be an issue here since we set it to `a.serverAddress`.